### PR TITLE
Rewrite Wallet RecyclerAssetList

### PR DESCRIPTION
**Pull request by wojtus7**
Rewrite logic to be more readable and fix some issues:
- Use RecyclerViewTypes to determine what component is rendered and it's height all in one place
- New LayoutAnimation logic (require `yarn` because need a patch-package to change things in RLV library itself)
- Rewrite rowRenderer to be clearer and use only logic from RecyclerViewTypes
- New fully rewritten auto-scroll logic to be more readable and fully generic (adding new things or changing sizes of components won't break it - I hope so 🙈)
- Remove all random ✨magic numbers ✨from RLV to avoid misalignments
- Fix first balance row margin issue
- When changing native currencies auto-scroll to top to avoid strange offset when entering wallet screen

### CODE REVIEW

**Comment by christianbaroni** [comment]
@wojtus7 this looks great functionally

only thing I noticed: the showcase section header is misaligned. see below (left is correctly aligned old version, right is new version)

Screen Shot 2020-04-29 at 4 12 37 PM

**Comment by wojtus7** [comment]
@christianbaroni fixed the showcase issue 
Didn't send a correct flag to the component 😅

File: src/components/asset-list/RecyclerAssetList.js [hunk 1]
```diff
@@ -562,317 +472,101 @@ class RecyclerAssetList extends Component {
...
+
+          const focusedFamilyItem = this.state.dataProvider._data[familyIndex]
+            .item;
+          const focusedFamilyHeight = ViewTypes.UNIQUE_TOKEN_ROW.calculateHeight(
+            {
+              amountOfRows: Math.ceil(
+                Number(focusedFamilyItem.childrenAmount) / 2
+              ),
+              isFirst: false,
+              isLast: false,
```
Comment (jinchung) [inline src/components/asset-list/RecyclerAssetList.js:505]: the new calculateHeight function for UNIQUE_TOKEN_ROW is not doing anything with the params isLast or paddingBottom

File: src/components/asset-list/RecyclerAssetList.js [hunk 2]
```diff
@@ -1014,89 +692,50 @@ class RecyclerAssetList extends Component {
...
-    if (type === ViewTypes.COIN_DIVIDER) {
-      return (
-        <CoinDivider
-          assetsAmount={item.assetsAmount}
-          balancesSum={item.value}
-          isCoinListEdited={isCoinListEdited}
-          nativeCurrency={this.props.nativeCurrency}
-        />
-      );
+    if (type.index === ViewTypes.HEADER.index) {
```
Comment (jinchung) [inline src/components/asset-list/RecyclerAssetList.js:701]: do we not need to include logic for if the index is ViewTypes.FOOTER ?

File: src/components/asset-list/RecyclerAssetList.js [hunk 3]
```diff
@@ -277,50 +271,76 @@ class RecyclerAssetList extends Component {
...
-            !sections[balancesIndex].data[firstBalanceIndex - 1]
-              .smallBalancesContainer
-          ) {
-            return ViewTypes.COIN_ROW_FIRST;
-          }
+
+          return {
+            height: ViewTypes.COIN_ROW.calculateHeight({
+              areSmallCollectibles: this.state.areSmallCollectibles,
+              isFirst:
```
Comment (jinchung) [inline src/components/asset-list/RecyclerAssetList.js:331]: can we pull out the isFirst logic on line 331 and line 338 as a separate const called isFirstCoinRow or something

File: src/components/asset-list/RecyclerAssetList.js (hunk 2 above)
Comment (wojtus7) [reply src/components/asset-list/RecyclerAssetList.js:701]: Not really - If the component doesn't have renderer rowRenderer doesn't have to return anything - for footer we wanted only to take some space - LayoutProvider above made that happen

We can just make rowRenderer return null for cleanliness

File: src/components/asset-list/RecyclerAssetList.js (hunk 3 above)
Comment (wojtus7) [reply src/components/asset-list/RecyclerAssetList.js]: Great idea 🎉
Applied

File: src/components/asset-list/RecyclerAssetList.js (hunk 1 above)
Comment (wojtus7) [reply src/components/asset-list/RecyclerAssetList.js]: Sorry - removed

**Pull request merged**